### PR TITLE
chore: add dependabot config

### PR DESCRIPTION
**Pull request by Texarkanine**
# Pull Request

## Summary

Add dependabot.yaml to get dependency update automation.

## Details

non-major dependencies are grouped, but split on dev/non-dev.
Major dependencies will always get updated independently.

## Testing

n/a

### Test Cases Covered

- [ ] Unit tests added/updated for new functionality
- [ ] Edge cases tested
- [ ] Error conditions handled
- [ ] User interaction flows verified

### Manual Testing

n/a

## Screenshots/Recordings

n/a

## Checklist

- [x] Commit message follows [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Commit message clearly describes the change
- [x] Includes breaking change notation if applicable

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          - "minor"
+          - "patch"
+    open-pull-requests-limit: 10
+    reviewers:
+      - "texarkanine"
+    assignees:
+      - "texarkanine"
+    commit-message:
+      prefix: "fix"
+      include: "scope"
```
[the remark this answers is not in the document]
Comment (Texarkanine) [reply .github/dependabot.yml]: > Dependency updates aren’t bug fixes.

Thats-where-youre-wrong-kiddo-meme-5-2943068678


I'm keeping `fix`.

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [18, 20, 22]
+        node-version: [18, 20, 22, 24]
```
[the remark this answers is not in the document]
Comment (Texarkanine) [reply .github/workflows/pr.yaml:16]: No, we stay on the oldest non-EOL'd LTS because we like our customers.
